tests/inference/testing_nodes/testing-pack/tools.py

```diff
@@ -0,0 +1,53 @@
+
+def MakeSmartType(t):
+    if isinstance(t, str):
+        return SmartType(t)
+    return t
+
+class SmartType(str):
+    def __ne__(self, other):
+        if self == "*" or other == "*":
+            return False
+        selfset = set(self.split(','))
+        otherset = set(other.split(','))
+        return not selfset.issubset(otherset)
```

Comment:
This seems to be backwards, because selftest will have more elements than otherset if self has multiple types and we are comparing with only one type.